docs/source/reference/kubernetes/multi-kubernetes.rst

```diff
@@ -0,0 +1,131 @@
+
+
+.. image:: /images/multi-kubernetes.svg
+
+
+Set Up Credentials for Multiple Kubernetes Clusters
+---------------------------------------------------
+
+To work with multiple Kubernetes clusters, you need to ensure you have the necessary credentials for each cluster.
+Check that your local ``~/.kube/config`` file has the credentials for each cluster. For setting up clusters and their credentials,
+see :ref:`kubernetes-setup-deploy`.
+
+For example, a ``~/.kube/config`` file may look like this:
+
+.. code-block:: yaml
+
+    apiVersion: v1
+    clusters:
+    - cluster:
+        certificate-authority-data: 
+        ...
+        server: https://xx.xx.xx.xx:45819
+      name: my-h100-cluster
+    - cluster:
+        certificate-authority-data:
+        ...
+        server: https://yy.yy.yy.yy:45819
+      name: my-tpu-cluster
+    contexts:
+    - context:
+        cluster: my-h100-cluster
+        user: my-h100-cluster
+      name: my-h100-cluster
+    - context:
+        cluster: my-tpu-cluster
+        namespace: my-namespace
+        user: my-tpu-cluster
+      name: my-tpu-cluster
+    current-context: my-h100-cluster
+    ...
+
+
+In this example, we have two Kubernetes clusters: ``my-h100-cluster`` and ``my-tpu-cluster``, and each Kubernetes cluster has a context for it.
+
+Point to a Kubernetes Cluster and Launch
+-----------------------------------------
+
+SkyPilot borrows the ``region`` concept from clouds to denote a Kubernetes cluster. You can point to a Kubernetes cluster
+by specifying the ``--region`` with the context name for that cluster.
+
+.. code-block:: console
+
+    # Check the GPUs available in a Kubernetes cluster
+    $ sky show-gpus --cloud kubernetes --region my-h100-cluster
+
+    Kubernetes GPUs (Context: my-h100-cluster)
+    GPU    QTY_PER_NODE            TOTAL_GPUS  TOTAL_FREE_GPUS  
+    H100   1, 2, 3, 4, 5, 6, 7, 8  8           8                
+
+    Kubernetes per node GPU availability
+    NODE_NAME                                 GPU_NAME  TOTAL_GPUS  FREE_GPUS  
+    my-h100-cluster-hbzn  H100      8           8
+    my-h100-cluster-w5x7  None      0           0
+
+When launching a SkyPilot cluster or task, you can also specify the context name with ``--region`` to launch the cluster or task in.
+
+.. code-block:: console
+
+    $ sky launch --cloud kubernetes --region my-tpu-cluster echo 'Hello World'
+
+
+.. note::
+
+    When you don't specify a region, SkyPilot will use the current context.
+
+
+Failover across Multiple Kubernetes Clusters
+--------------------------------------------
+
+SkyPilot enables you to failover across multiple Kubernetes clusters. It is useful when you want to launch a task in any of the clusters with available GPUs.
+
+Different from cloud providers, SkyPilot does not failover through different regions (contexts) by default, because multiple
```

Comment:
```suggestion
Unlike clouds, SkyPilot does not failover through different Kubernetes clusters (regions) by default because each Kubernetes clusters can have a different purpose. 
```